src/microbe_stage/SpawnSystem.cs

```diff
@@ -232,7 +235,19 @@ private void SpawnEntities(Vector3 playerPosition, Vector3 playerRotation, int e
 
         int spawned = 0;
 
+        attemptsPerSpawnType.Clear();
+        int maxAttempts = -1;
         foreach (var spawnType in spawnTypes)
+        {
+            int numAttempts = Math.Min(Math.Max(spawnType.SpawnFrequency * 2, 1),
```

Comment:
I know you used existing code as a basis, but as this is being rewritten anyway, I'd like to see this variable to be renamed to just `attempts`

Reply:
I also tend to use `num` prefix for number variables, as it makes clear, in my opinion, that we are counting things. I can change it if this fits the project coding style better, though.

Reply:
Yes please. Abbreviations (except a few) are banned in the styleguide. Also this is the only source file that has "num" prefix in a name.

Reply:
Indeed, I use it so often I did not even realize it was an abbreviation. My bad, it's fixed!